File: be/src/main/java/ds/project/orino/common/response/exception/ErrorCode.java
package ds.project.orino.common.response.exception;

import org.springframework.http.HttpStatus;

public enum ErrorCode {

    // GLOBAL
    BAD_REQUEST("GLB-ERR-001", "잘못된 요청입니다.", HttpStatus.BAD_REQUEST),
    METHOD_NOT_ALLOWED("GLB-ERR-002", "허용되지 않은 메서드입니다.", HttpStatus.METHOD_NOT_ALLOWED),
    INTERNAL_SERVER_ERROR("GLB-ERR-003", "내부 서버 오류입니다.", HttpStatus.INTERNAL_SERVER_ERROR),

    // AUTH
    INVALID_CREDENTIALS("AUTH-ERR-001", "아이디 또는 비밀번호가 올바르지 않습니다.", HttpStatus.UNAUTHORIZED),
    INVALID_TOKEN("AUTH-ERR-002", "유효하지 않은 토큰입니다.", HttpStatus.UNAUTHORIZED);

    private final String code;
    private final String message;
    private final HttpStatus httpStatus;

    ErrorCode(String code, String message, HttpStatus httpStatus) {
        this.code = code;
        this.message = message;
        this.httpStatus = httpStatus;
    }

    public String getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }

    public String getMessage(Throwable e) {
        return message + " - " + e.getMessage();
    }

    public HttpStatus getHttpStatus() {
        return httpStatus;
    }

    public String getMessage(String additionalMessage) {
        if (additionalMessage != null && !additionalMessage.isBlank()) {
            return message + " - " + additionalMessage;
        }
        return message;
    }
}

File: be/src/main/kotlin/ds/project/orino/common/response/exception/ErrorCode.kt
package ds.project.orino.common.response.exception

enum class ErrorCode(val code: String, val message: String) {

    // GLOBAL
    BAD_REQUEST("GLB-ERR-001", "잘못된 요청입니다."),
    METHOD_NOT_ALLOWED("GLB-ERR-002", "허용되지 않은 메서드입니다."),
    INTERNAL_SERVER_ERROR("GLB-ERR-003", "내부 서버 오류입니다.");

    fun getMessage(e: Throwable): String {
        return "$message - ${e.message}"
    }

    fun getMessage(additionalMessage: String?): String {
        return additionalMessage?.takeIf { it.isNotBlank() }?.let {
            "$message - $it"
        } ?: message
    }
}